src/main/java/io/vertx/ext/couchdb/CouchdbCredentials.java

```diff
@@ -0,0 +1,75 @@
+package io.vertx.ext.couchdb;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.codegen.json.annotations.JsonGen;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.authentication.Credentials;
+import io.vertx.ext.auth.authentication.UsernamePasswordCredentials;
+import io.vertx.ext.auth.authentication.TokenCredentials;
+
+@DataObject
+@JsonGen
+public class CouchdbCredentials {
+
```

Comment:
Yes. The reason of why I added a new `CouchdbCredentials` was because I am using `@JsonGen` that can generate a `CouchdbClientOptionsConverter `class which can be used for converting from and to json like this:

```
public CouchdbClientOptions(JsonObject json) {
    this();
    CouchdbClientOptionsConverter.fromJson(json, this);
  }
```
However the `CouchdbClientOptionsConverter `class expects all the fields of the `CouchdbClientOptions` to be a concrete class thus I cannot use the `io.vertx.ext.auth.authentication.Credentials` because it's an interface. But please let me know if you have any ideas around this or if we still want to generate a converter class in the first place.

Reply:
would transient work? Also interesting question: What options beyond the webclient options do we want to introduce?